### PR TITLE
Update ruby version to 3.3

### PR DESCRIPTION
**Pull request by shopiwags**
## Context

- Updating the ruby version to 3.2
- Updating the gemfile references for versions lower than 3.2 to `rails-edge`

## Tophat

I imported this branch into a spin instance:
`gem "payment_icons", :github => "activemerchant/payment_icons", :branch => "ruby-version-update"`

Ran through a checkout and observed the icon's loading:

Screenshot 2024-10-01 at 2 28 17 PM

### CODE REVIEW

**Review by jadnhm — APPROVED** [review]

File: .github/workflows/ruby-ci.yml [hunk 1]
```diff
@@ -17,26 +17,20 @@ jobs:
...
-          - version: 3.0
-            gemfile: gemfiles/rails-6-0.gemfile
           - version: 3.1
-            gemfile: gemfiles/rails-6-0.gemfile
+            gemfile: gemfiles/rails-edge.gemfile
           - version: 3.2
-            gemfile: gemfiles/rails-6-0.gemfile
+            gemfile: gemfiles/rails-7-0.gemfile
+          - version: 3.3
+            gemfile: gemfiles/rails-7-0.gemfile
```
Comment (danmo91) [inline .github/workflows/ruby-ci.yml:33]: I don't have context on Rails and Ruby version compatibility. This [compatibility chart](https://www.fastruby.io/blog/ruby/rails/versions/compatibility-table.html) suggests that Ruby 3.2 and Ruby 3.3 are compatible with Rails 7.

I can [confirm](https://github.com/rails/rails/blob/main/rails.gemspec#L12) that Ruby 3.1 support was dropped on the latest rails main 

image

**Review by danmo91 — APPROVED** [review]
I tophatted this change on checkout classic and LGTM!

| Syntax      | Description |
| ----------- | ----------- |
| Checkout Classic | Payment methods      | image       |
| Thank you page   | image  |
| Order Confirmation Email   |   image    |

File: .github/workflows/ruby-ci.yml (hunk 1 above)
Comment (larouxn) [reply .github/workflows/ruby-ci.yml:33]: Might be worth trying to remove the `exclude` block altogether. It was for incompatible Ruby + Rails combos but not many exist at this point other than Ruby 3.1 + Rails 8 (not yet out, so, main).

Mostly curious why we need Ruby 3.2 and 3.3 + Rails 7.

**Review by larouxn — APPROVED** [review]
Thank you 🙏

**Pull request merged**